filter.cc

```diff
@@ -34,26 +35,32 @@ Filter::Filter() :
   dataLen(-1),
   domainList(nullptr),
   host(nullptr),
-  domainCount(0),
-  antiDomainCount(0),
-  hostLen(-1) {
+  hostLen(-1),
+  domains(nullptr),
+  antiDomains(nullptr),
+  domainsParsed(false) {
 }
 
 Filter::~Filter() {
-  if (borrowed_data) {
-    return;
-  }
-  if (data) {
-    delete[] data;
+  if (domains) {
```

Comment:
`domains` and `antiDomains` are lazily created, so check if we need to delete them.